### PR TITLE
feat: Dockerfile tweaks

### PR DESCRIPTION
**Pull request by kik-kik**
# feat: Dockerfile tweaks

List of changes:
- Added `.vscode` to `.gitignore`
- `.dockerignore` added to exclude some files from the image which I believe should not be needed
- Updated deprecated `MAINTAINER` tag to `LABEL`
- Making sure `no-cache` flags are passed when installing packages
- Condensed Docker commands into fewer commands
- Commented out (for now) 3 OS packages, image builds fine. Could those be needed for anything else?

### CODE REVIEW

File: .Dockerignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
+### A list of files and file patterns to be excluded from the Dockerfile
+
+# 3rd-party configuratiojn
```
Comment (scholtzan) [inline .Dockerignore:3]:
```suggestion
# 3rd-party configuration
```

File: Dockerfile [hunk 2]
```diff
@@ -9,31 +10,43 @@ RUN groupadd --gid ${USER_ID} ${GROUP_ID} && \
...
-RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-key C99B11DEB97541F0
-RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-key C99B11DEB97541F0
-RUN apt-add-repository https://cli.github.com/packages
-RUN apt update
-RUN apt install -y gh
+RUN apt-get update -qqy \
+    && apt-get install --no-install-recommends -qqy \
+        # python-dev \
+        # build-essential \
+        # curl \
```
Comment (scholtzan) [inline Dockerfile:17]: I think we still need those, for example for this: https://github.com/mozilla/lookml-generator/blob/df244e9a545a057776f15d277e63db8f7d9a1172/bin/generate#L235
```suggestion
        python-dev \
        build-essential \
        curl \
```

File: .Dockerignore [hunk 3]
```diff
@@ -0,0 +1,21 @@
+### A list of files and file patterns to be excluded from the Dockerfile
```
Comment (relud) [inline .Dockerignore:1]: I just tested it, and at least for me this file needs to be `.dockerignore` (all lower case)

**Review by relud — DISMISSED** [review]
r+wc

File: .Dockerignore (hunk 1 above)
Comment (kik-kik) [reply .Dockerignore]: oh no, me and my typos 🙈

File: .Dockerignore (hunk 3 above)
Comment (kik-kik) [reply .Dockerignore]: yes, you are right. I thought I renamed it, but I guess I did not. Should be fixed now.

File: Dockerfile (hunk 2 above)
Comment (kik-kik) [reply Dockerfile:17]: do we need `python-dev` and `build-essential` though? Is there any way for us to test this?

**Review by scholtzan — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (scholtzan) [reply Dockerfile]: Turns out they are used for `ssh-keyscan`. Adding them back

**Pull request merged**